### ISSUE TITLE
sts / download from s3 fails on Python

### ISSUE THREAD

**Issue by aviramha** [metalbear-co/mirrord#1747]
Reproducing:

requirements:
```
django==4.2.3
gunicorn==20.1.0
django-storages==1.12.3
djangorestframework==3.14.0
django-redis==5.2.0
psycopg2-binary==2.9.3
requests==2.31.0
django-extensions==3.1.5
django-grappelli==3.0.3
whitenoise==6.1.0
urllib3==1.26.9
Pillow==9.5.0
pycurl==7.45.2
celery==5.2.6
django-multiselectfield==0.1.12
httpagentparser==1.9.2
raven==6.10.0
gevent==21.12.0
pandas==1.4.4
django-celery-beat==2.5.0
user-agents==2.2.0
pytz==2023.3
ipython==8.10
boto3==1.21.22
botocore==1.24.46
```
Python 3.10.11 on macOS M1

This code:
```
import ssl
context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
context.load_verify_locations('/etc/ssl/certs/ca-certificates.crt')
```

with this mirrord.json:
```

{
    "feature": {
        "network": {
            "incoming": "mirror",
            "outgoing": true
        },
        "fs": {"mode": "read", "read_only": ["/etc/ssl/certs/ca-certificates.crt"]},
        "env": true
    }
}

```

**Comment by aviramha** [comment]
Was missing hook to `read$NOCANCEL` (we had `_read$NOCANCEL` ;( )

**Comment by aviramha** [comment]
Closed in #1777
